content/best_practices.adoc

```diff
@@ -56,7 +56,48 @@ Some operations require logging/auditing to allow to understand what happened du
+The reviewer must try to detect if the fix/feature could have significant bad impacts on the performances of the application. Performances issues can be expensive to find and fix, so it is important to raise any concern on this topic at this stage. Here are some examples of performance issues causes: too many database requests, slow database query, missing index in the database, too many HTTP requests, …
+
+=== Front-End Performances
+
+​In Front-End developing, it's important to consider best solutions that reduces Memory & CPU consumption in Browser. In fact, a Web page executes only one single thread for all Vue apps, so it's important to avoid adding a lot of stuff to do when mounting a Vue application (https://www.red-gate.com/simple-talk/development/dotnet-development/javascript-single-threaded/[Read more about JS lifecycle]). In fact, when adding a heavy computing statements in page loading phase, it will *pause* mounting all other Vue applications until the computing is finished, because of single-threaded execution architecture of Web pages. Consequently, having one Vue app that isn't sufficiently performant in a web page, especially for a TopBar application (that is present in all pages), will degrade the rendering performances of the whole page.
+
+Knowing that, the Vue applications rendering has to be made in an asynchrnous way without impacting other applications present in the page. For example, when having information to retrieve from Server-End using REST API calls, a fetch call will be used. To ensure to have a good UX, a loading effect in the application itself has to be rendered first time the Vue application is mounted, until all needed information are retrieved.
+
+But, using the fetch API to retrieve information asynchronously from Server-End has a big drawback. In fact, a Browser allows a limited concurrent HTTP requests for each domain (https://blog.stackpath.com/glossary-domain-sharding/[Domain Sharding]). In a web page, that holds a lot of Vue applications, each one will retrieve information from Server-End using different REST endpoints. This can consequently lead to a very bad UX with a considerable idle time for HTTP requests. Fortunately, the Web Standards has evolved to take into consideration this complexity of new Web Applications especially by introducing the `HTTP/2` protocol (https://www.limelightonline.co.nz/blog/what-http2-benefit-your-website/[Read more about benifits]). To benefit from it, and to allow to increase the parallel concurrent requests to send to Server-End, we will have to `preload` and `prefetch` mandatory information that renders the Web Page (https://medium.com/reloading/preload-prefetch-and-priorities-in-chrome-776165961bbf[Read more]).
```

Comment:
```suggestion
Knowing that the Vue applications rendering has to be made in an asynchronous way without impacting other applications present on the page. For example, when having information to retrieve from Server-End using REST API calls, a fetch call will be used. To ensure to have a good UX, a loading effect in the application itself has to be rendered the first time the Vue application is mounted, until all needed information is retrieved.

But, using the fetch API to retrieve information asynchronously from Server-End has a big drawback. In fact, a Browser allows limited concurrent HTTP requests for each domain (https://blog.stackpath.com/glossary-domain-sharding/[Domain Sharding]). In a web page, that holds a lot of Vue applications, each one will retrieve information from Server-End using different REST endpoints. This can consequently lead to a very bad UX with a considerable idle time for HTTP requests. Fortunately, the Web Standards has evolved to take into consideration this complexity of new Web Applications especially by introducing the `HTTP/2` protocol (https://www.limelightonline.co.nz/blog/what-http2-benefit-your-website/[Read more about benifits]). To benefit from it, and to allow to increase the parallel concurrent requests to send to Server-End, we will have to `preload` and `prefetch` mandatory information that renders the Web Page (https://medium.com/reloading/preload-prefetch-and-priorities-in-chrome-776165961bbf[Read more]).
```